### PR TITLE
[Connector API] Implement update scheduling action

### PR DESCRIPTION
**Pull request by jedrazb**
## Changes

- Implement `PUT _connector/{connector_id}/scheduling` endpoint
- This endpoint expects body with a new connector scheduling object, it overwrites the existing `scheduling` field in a given connector doc

## Verification
- Add unit tests for index service
- Add BWC tests
- Add YML tests

Also, verified new endpoint manually.

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/api/connector.update_scheduling.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        "application/json"
+      ],
+      "content_type": [
+        "application/json"
+      ]
+    },
+    "url": {
+      "paths": [
+        {
+          "path": "/_connector/{connector_id}/scheduling",
```
Comment (carlosdelest) [inline rest-api-spec/src/main/resources/rest-api-spec/api/connector.update_scheduling.json:21]: I'd _say_ that actions done on a resource in ES tend to have underscore:

```suggestion
          "path": "/_connector/{connector_id}/_scheduling",
```

I think that's more in line with what we have. WDYT @kderusso ? (Or anyone that can keep me honest here)

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java [hunk 2]
```diff
@@ -166,6 +169,36 @@ public void onFailure(Exception e) {
...
+     * Updates the {@link ConnectorScheduling} property of a {@link Connector}.
+     *
+     * @param request  The request for updating the connector's scheduling.
+     * @param listener The listener for handling responses, including successful updates or errors.
+     */
+    public void updateConnectorScheduling(UpdateConnectorSchedulingAction.Request request, ActionListener<UpdateResponse> listener) {
+        try {
+            String connectorId = request.getConnectorId();
+            final UpdateRequest updateRequest = new UpdateRequest(CONNECTOR_INDEX_NAME, connectorId).doc(
+                new IndexRequest(CONNECTOR_INDEX_NAME).opType(DocWriteRequest.OpType.INDEX)
```
Comment (carlosdelest) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java:182]: I'm not sure if this applies as it's part of the update request, but I think the proper operation should be update. 

```suggestion
                new IndexRequest(CONNECTOR_INDEX_NAME).opType(DocWriteRequest.OpType.UPDATE)
```

Let's try this with implementing the 404 test mentioned above and see if it works correctly 👍

**Review by carlosdelest — APPROVED** [review]
Good job! You got the hang of it 🚀 

A few comments about testing and paths that should be considered. They can be done in a follow up PR.

File: x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/331_connector_update_scheduling.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+"Update Connector Scheduling - 400 status code returned when required fields are missing":
+  - do:
+      catch: "bad_request"
+      connector.update_scheduling:
+        connector_id: test-connector
+        body:
+          incremental:
+            enabled: false
+            interval: 3 0 0 * * ?
```
[the remark this answers is not in the document]
Comment (jedrazb) [reply x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/331_connector_update_scheduling.yml:57]: Yeah that makes sense, let me add the test in this PR :)

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java (hunk 2 above)
Comment (jedrazb) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java:182]: That's what I thought as well - but the IndexRequest doesn't allow to `OpType.UPDATE` ([IndexRequest source code](https://github.com/elastic/elasticsearch/blob/714aa680f11514ee981682785423bc4d3bb7d383/server/src/main/java/org/elasticsearch/action/index/IndexRequest.java#L216)) - My understanding is that we are creating a new scheduling object (overwriting a field in a parent connector doc)

Comment (carlosdelest) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java:182]: Thanks for confirming! 👍

File: x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/331_connector_update_scheduling.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  - match:
+      status: 400
```
[the remark this answers is not in the document]
Comment (jedrazb) [reply x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/331_connector_update_scheduling.yml:60]: Yes will add 👍

**Comment by jedrazb** [comment]
@carlosdelest I added more tests testing for 404s and 400 with malformed cron expression. Also updated the API path to`_scheduling`. Thank you for quick feedback! :)

**Pull request merged**